### PR TITLE
Show exact ai score when exact-match confidence is high

### PR DESCRIPTION
**Pull request by davidsbailey**
Depends on https://github.com/code-dot-org/code-dot-org/pull/57078. Finishes https://codedotorg.atlassian.net/browse/AITT-489.

Done in this PR: 
* UI updates: use exact-match confidence to determine whether to show exact-match evidence level. when we do:
  * highlight only one "suggested" evidence level bubble
  * update text describing evidence level to name only one evidence level
  * make sure we are showing exact-match confidence, not pass-fail confidence
* refactoring / cleanup: 
  * renamed ai_confidence field to aiConfidencePassFail in the json that is passed to the client. this avoids some ambiguity on the client since aiConfidence is then used to represent whichever confidence value is to be shown
  * variable renames and other minor changes
  * [remove rating text from css classnames](https://github.com/code-dot-org/code-dot-org/pull/57125/commits/36fe969e82ca7074e442e81d9358fef0bcb263b1)
  * converted one existing test to React Testing Library

### before
before--aitt-basic-updates-to-ui-rubric-component

### after 

after-exact-extensive

after-exact-no-ev

## Testing story

* New RTL unit tests covering UI updates above
* end-to-end manual testing

## Follow-up work

For testing, I used the `2024-03-06-ai-rubrics-json-reason-confidence-exact` release in S3, which is the same as the `ai-rubrics-json-reason` release but with confidence-exact.json files added to it. The follow-up work to enable this feature is to ship a release containing these `confidence-exact.json` files.

### CODE REVIEW

File: apps/src/templates/rubrics/AiAssessmentBox.jsx [hunk 1]
```diff
@@ -25,16 +26,22 @@ export default function AiAssessmentBox({
...
     return i18n.aiStudentAssessment({
       studentName: studentName,
       understandingLevel: assessment,
     });
   };
 
+  const getStudentAssessmentString = () => {
+    if (aiEvalInfo.showExactMatch) {
+      return UNDERSTANDING_LEVEL_STRINGS[aiUnderstandingLevel];
+    }
```
Comment (davidsbailey) [inline apps/src/templates/rubrics/AiAssessmentBox.jsx:42]: this code limits the text we display to a single evidence level

File: apps/src/templates/rubrics/EvidenceLevelsForTeachersV2.jsx [hunk 2]
```diff
@@ -28,16 +29,25 @@ export default function EvidenceLevelsForTeachersV2({
...
       if (understanding !== INVALID_UNDERSTANDING || !isAiAssessed) {
         return [];
       }
-      if (aiEvalInfo.understanding > 1) {
-        return [2, 3];
-      } else if (aiEvalInfo.understanding >= 0) {
-        return [0, 1];
+      if (aiEvalInfo.showExactMatch) {
+        return [aiEvalInfo.understanding];
+      }
```
Comment (davidsbailey) [inline apps/src/templates/rubrics/EvidenceLevelsForTeachersV2.jsx:41]: This code limits us to highlighting one suggested evidence level bubble

File: apps/src/templates/rubrics/LearningGoals.jsx [hunk 3]
```diff
@@ -221,6 +224,13 @@ export default function LearningGoals({
 
   const aiEvalInfo = getAiInfo(learningGoals[currentLearningGoal].id);
 
+  let aiConfidence;
+  if (aiEvalInfo) {
+    aiConfidence = aiEvalInfo.showExactMatch
+      ? aiEvalInfo.aiConfidenceExactMatch
+      : aiEvalInfo.aiConfidencePassFail;
+  }
```
Comment (davidsbailey) [inline apps/src/templates/rubrics/LearningGoals.jsx:242]: this change updates aiConfidence to depend on whether we are showing pass-fail or exact-match AI scores. all uses of aiConfidence in client code should now refer to this computed value, not the `ai_confidence` value on the server which is actually the pass-fail confidence level.

File: apps/test/unit/templates/rubrics/LearningGoalsTest.jsx [hunk 4]
```diff
@@ -65,6 +69,136 @@ describe('LearningGoals', () => {
...
+          learningGoals={learningGoals}
+          aiEvaluations={aiEvaluations}
+          studentLevelInfo={studentLevelInfo}
+          teacherHasEnabledAi
+          canProvideFeedback
+        />
+      );
+      screen.getByText(
+        'Stella has achieved Extensive or Convincing Evidence for this learning goal.'
+      );
```
Comment (Nokondi) [inline apps/test/unit/templates/rubrics/LearningGoalsTest.jsx:169]: Do you need an expect here?

File: apps/test/unit/templates/rubrics/LearningGoalsTest.jsx [hunk 5]
```diff
@@ -65,6 +69,136 @@ describe('LearningGoals', () => {
...
+      render(
+        <LearningGoals
+          learningGoals={learningGoals}
+          aiEvaluations={aiEvaluations}
+          studentLevelInfo={studentLevelInfo}
+          teacherHasEnabledAi
+          canProvideFeedback
+        />
+      );
+      screen.getByText('AI Confidence: medium');
```
Comment (Nokondi) [inline apps/test/unit/templates/rubrics/LearningGoalsTest.jsx:181]: Same here (see previous comment)

File: apps/test/unit/templates/rubrics/LearningGoalsTest.jsx [hunk 6]
```diff
@@ -65,6 +69,136 @@ describe('LearningGoals', () => {
...
+      render(
+        <LearningGoals
+          learningGoals={learningGoals}
+          aiEvaluations={myAiEvaluations}
+          studentLevelInfo={studentLevelInfo}
+          teacherHasEnabledAi
+          canProvideFeedback
+        />
+      );
+      screen.getByText(
```
Comment (Nokondi) [inline apps/test/unit/templates/rubrics/LearningGoalsTest.jsx:124]: Do you need an expect here?

**Review by Nokondi — COMMENTED** [review]
Looks great! Just had a question about using expect in some of your tests. The comments you added here are very helpful - might also be useful to include some of them in the code. (I say this, but I'm terrible at remembering to document my own code 🤣 )

File: apps/test/unit/templates/rubrics/LearningGoalsTest.jsx (hunk 4 above)
Comment (davidsbailey) [reply apps/test/unit/templates/rubrics/LearningGoalsTest.jsx:169]: I believe the answer is no! admittedly I had to look this up to know the answer... but it looks like RTL "queries" starting with "get" will throw if the element is not found: https://testing-library.com/docs/queries/about/

Comment (davidsbailey) [reply apps/test/unit/templates/rubrics/LearningGoalsTest.jsx:169]: I was able to confirm that all 3 of the tests using "get" queries do fail if I ask them to look for a nonexistent string.

**Comment by davidsbailey** [comment]
Thanks for the feedback, Mark! I've included some updated code comments mirroring the notes in my inline PR comments.

File: apps/test/unit/templates/rubrics/LearningGoalsTest.jsx (hunk 4 above)
Comment (Nokondi) [reply apps/test/unit/templates/rubrics/LearningGoalsTest.jsx:169]: OK, good to know!

**Review by Nokondi — APPROVED** [review]
Great work!

**Pull request merged**